component-runtime-manager/src/main/java/org/talend/sdk/component/runtime/manager/ComponentManager.java

```diff
@@ -621,26 +633,38 @@ private static Path findDefaultM2() {
                 .get(System
                         .getProperty("talend.component.manager.m2.settings",
                                 System.getProperty("user.home") + "/.m2/settings.xml"));
+        log.debug("[findDefaultM2] searching for localRepository location in {}", settings);
         if (java.nio.file.Files.exists(settings)) {
             try {
-                // faster to do that than previous code (commented after)
-                final String content = new String(java.nio.file.Files.readAllBytes(settings), StandardCharsets.UTF_8);
-                final int start = content.indexOf("<localRepository>");
                 String localM2RepositoryFromSettings = null;
-                if (start > 0) {
-                    final int end = content.indexOf("</localRepository>", start);
-                    if (end > 0) {
-                        localM2RepositoryFromSettings = content.substring(start + "<localRepository>".length(), end);
+                // do not introduce a xml parser so will do with what we have...
+                final String content = new String(java.nio.file.Files.readAllBytes(settings), StandardCharsets.UTF_8);
+                final Pattern comments = Pattern.compile("(<!--.*?-->)", Pattern.DOTALL);
+                final Pattern emptyLines = Pattern.compile("^\\s*$|\\n|\\r\\n");
```

Comment:
not an expert here
But maybe we want to include \\r too. What do you think? (Idea says that \\r is line separator in Classic Macos that's why I think it may exist without following \\n)

Reply:
Also, I tried to check it on https://regex101.com/ (mode \gm)
And it doesn't match whitespaces

Reply:
We may add the `\r` but I doubt  that anyone is still using MacOS classic with a ppc :-D 
The goal of `enptyLines` is to flatten file to one big line like this 
```xml
<?xml version="1.0" encoding="UTF-8"?><settings xmlns="http://maven.apache.org/SETTINGS/1.2.0"    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"    xsi:schemaLocation="http://maven.apache.org/SETTINGS/1.2.0 http://maven.apache.org/xsd/settings-1.2.0.xsd">        <localRepository>/home/zorro80</localRepository>                <pluginGroups>            </pluginGroups>        <proxies>            </proxies>        <servers>                    </servers>        <mirrors>                <mirror>            <id>maven-default-http-blocker</id>            <mirrorOf>external:http:*</mirrorOf>            <name>Pseudo repository to mirror external repositories initially using HTTP.</name>            <url>http://0.0.0.0/</url>            <blocked>true</blocked>        </mirror>    </mirrors>        <profiles>                    </profiles>    </settings>
```


Its goal is not to reduce extra space. It's done w/ the trim() call on successful capture. 
`^\\s*$` is to match empty line containing 0 or more spaces.

Reply:
Oh, I see, thank you!
(missed $ at the end of the first part)